### DISCUSSION TITLE
can we use empty array name space in withTranslation Hoc

### DISCUSSION THREAD

**Discussion by chaitanya71998** [i18next/i18next#1937] [Q&A]
can we use with an empty array when using withTranslation for name space what happens on this
```js
const Wrapped = withTranslation([], { withRef: true })(MyComponent);
```

**Comment by jamuhl** [comment]
In that case I guess you will get an unbound `t` function. It will not assert namespaces are loaded. Defacto you can completely remove that withTranslation wrapper and directly use `i18n.t`

**Reply by chaitanya71998** [reply]
https://codesandbox.io/s/react-i18n-demo-forked-psoty7?file=/src/index.tsx like in this example i am using class components. so to use t function i must use withTranslation and i need not want to restrict translation to particular namespace. so i was thinking like on giving empty array, i18n check the keys given in t function to particular name space given to it like below?
also, will the t function didn't find a key in the name space that was given as argument to HOC like below will it check in other namespaces?
```js
class MyComponent {
render(){
const {t} = this.props
return <>(t("intro:welcome")</>
}
}
export default withTranslation('translation, { withRef: true })(MyComponent);

```

**Reply by jamuhl** [reply]
you do never restrict the t function - you say what namespaces need to be loaded when rendering that component.

first namespace in array will be default and can be accessed without prepending the namespace separated by namespace delimiter `ns:key`

**Reply by chaitanya71998** [reply]
got it.
